### PR TITLE
Alternative to genpop PR, adds genpop interface to our current brig timer

### PR DESCRIPTION
**Pull request by maxion12345**
# Document the changes in your pull request
TODO:

- [x] Fix security hud not updating when you brig someone (Sets their record to incarcerated but HUD won't update until something forces it (removing mask, etc)

kinda realized that our current brig timer does 80% of the things the genpop one does

So, why merge this over the port?
No funky IDs to fuck around with
No turnstiles on brig doors to get yourself trapped in
[NEW] you can do automatic record-setting
[NEW] swipe prisoner ID to set their name automatically if you can't be fucked to type it
literally two button operation

Why merge genpop over this?
Easy to expand brig system
And you can stuff more then one prisoner into a cell

oh I had to remove the random brig cell variants because that broke the brig lockers automatically closing
and here's a photo of the UI

# Wiki Documentation

add a note near brig procedures about swiping the prisoner's ID to automatically set the name for automatic record setting

# Changelog

Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR.
If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name.

:cl:  
rscadd: Added preset crime buttons to brig timers
rscdel: Removed brig cell variations due to breaking automatic locker locking 
rscadd: Turnstiles to brig
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
I liked the auto adding to ID's :'( Maybe let us put their ID in there and itll auto add the crime?

**Comment by maxion12345** [comment]
> I liked the auto adding to ID's :'( Maybe let us put their ID in there and itll auto add the crime?

I'll see if I can port the auto-record setting from genpop

**Comment by maxion12345** [comment]
ready for testmerge

**Review by patpol4 — APPROVED** [review]
This looks really bad on box with the half door but ok

**Review by JamieD1 — APPROVED** [review]

**Review by adamsong — DISMISSED** [review]
Why is this touching rust_g?

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
tgui lookover 2 seconds

File: code/game/machinery/doors/brigdoors.dm [hunk 1]
```diff
@@ -63,6 +104,13 @@
 		obj_break()
 	update_icon()
 
+/obj/machinery/door_timer/attackby(obj/item/W, mob/user, params)
+	var/obj/item/card/id/card = W.GetID()
+	if (card)
+		say("Prisoner name set.")
+		desired_name = card.registered_name
+	else
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm:112]: empty return?

File: code/game/machinery/doors/brigdoors.dm [hunk 2]
```diff
@@ -73,12 +121,13 @@
 
 	if(timing)
 		if(world.time - activation_time >= timer_duration)
-			timer_end() // open doors, reset timer, clear status screen
+			timer_end()
+			 // open doors, reset timer, clear status screen
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm]: why

File: code/game/machinery/doors/brigdoors.dm [hunk 3]
```diff
@@ -97,6 +146,19 @@
...
+	if(desired_crime)
+		var/datum/data/record/R = find_record("name", desired_name, GLOB.data_core.security)
+		if(R)
+			R.fields["criminal"] = "Incarcerated"
+			var/crime = GLOB.data_core.createCrimeEntry(desired_crime, null, user.real_name, station_time_timestamp())
+			GLOB.data_core.addCrime(R.fields["id"], crime)
+			investigate_log("New Crime: <strong>[desired_crime]</strong> | Added to [R.fields["name"]] by [key_name(user)]", INVESTIGATE_RECORDS)
+			say("Criminal record for [R.fields["name"]] successfully updated with inputted crime.")
+			playsound(loc, 'sound/machines/ping.ogg', 50, 1)
+		else if(desired_name == null)
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm]:
```suggestion
		else if(!desired_name)
```

File: code/game/machinery/doors/brigdoors.dm [hunk 4]
```diff
@@ -237,6 +318,17 @@
 			. = set_timer(preset_time)
 			if(timing)
 				activation_time = world.time
+		if("prisoner_name")
+			var/prisoner_name = stripped_input(usr, "Input prisoner's name...", "Crimes", desired_name)
+			if(prisoner_name == null | !Adjacent(usr))
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm]: I doubt they care about short-circuiting logic here
```suggestion
			if(!prisoner_name || !Adjacent(usr))
```

File: code/game/machinery/doors/brigdoors.dm [hunk 5]
```diff
@@ -219,8 +296,12 @@
 				. = set_timer(time_left()+value)
 		if("start")
 			timer_start()
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm]:
```suggestion
			timer_start(usr)
```

File: code/game/machinery/doors/brigdoors.dm [hunk 6]
```diff
@@ -237,6 +318,17 @@
...
 			if(timing)
 				activation_time = world.time
+		if("prisoner_name")
+			var/prisoner_name = stripped_input(usr, "Input prisoner's name...", "Crimes", desired_name)
+			if(prisoner_name == null | !Adjacent(usr))
+				return FALSE
+			desired_name = prisoner_name
+		if("presetCrime")
+			var/value = text2num(params["preset"])
+			var/preset_crime = params["crime"]
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm]: be aware this looks open to manipulating crime names clientside, and side this isn't filtered you can display random stuff in the crimes.
Should probably validate that this crime is one of the actual crimes

File: code/game/machinery/doors/brigdoors.dm [hunk 7]
```diff
@@ -237,6 +318,17 @@
 			. = set_timer(preset_time)
 			if(timing)
 				activation_time = world.time
+		if("prisoner_name")
+			var/prisoner_name = stripped_input(usr, "Input prisoner's name...", "Crimes", desired_name)
+			if(prisoner_name == null | !Adjacent(usr))
+				return FALSE
+			desired_name = prisoner_name
+		if("presetCrime")
+			var/value = text2num(params["preset"])
```
Comment (Bibby0110) [inline code/game/machinery/doors/brigdoors.dm:326]: open to manipulating but the timer is clamped so it doesn't matter

**Review by Bibby0110 — APPROVED** [review]
can't be bothered to look at tgui nevermind, do this though

**Comment by patpol4** [comment]
So is this good to go?

**Comment by maxion12345** [comment]
> So is this good to go?

aye

**Pull request merged**